src/Console/Command/CompileCommand.php

```diff
@@ -0,0 +1,236 @@
+<?php
+
+namespace Castor\Console\Command;
+
+use Castor\PathHelper;
+use Castor\PlatformUtil;
+use Symfony\Component\Console\Command\Command;
+use Symfony\Component\Console\Helper\ProgressBar;
+use Symfony\Component\Console\Input\ArrayInput;
+use Symfony\Component\Console\Input\InputInterface;
+use Symfony\Component\Console\Input\InputOption;
+use Symfony\Component\Console\Output\OutputInterface;
+use Symfony\Component\Console\Style\SymfonyStyle;
+use Symfony\Component\Filesystem\Filesystem;
+use Symfony\Component\HttpClient\HttpClient;
+use Symfony\Component\Process\Process;
+use Symfony\Contracts\HttpClient\HttpClientInterface;
+
+/**
+ * @internal
+ */
+class CompileCommand extends Command
+{
+    private readonly HttpClientInterface $httpClient;
+
+    public function __construct(
+        readonly private Filesystem $fs = new Filesystem()
+    ) {
+        parent::__construct();
+
+        $this->httpClient = HttpClient::create();
+    }
+
```

Comment:
Sorry, I was not clear enough. Cound you inject the $fs, and the $httpclient declared in the application factory?